core/src/commonMain/kotlin/com/powersync/utils/Json.kt

```diff
@@ -0,0 +1,15 @@
+package com.powersync.utils
+
+import kotlinx.serialization.json.Json
+
+/**
+ * A global instance of a JSON serializer.
+ */
+object JsonUtil {
+    val json = Json {
+        encodeDefaults = true
+        isLenient = true
```

Comment:
Is there a reason for this to be true? As far as I understand it's only "invalid" JSON (unquoted keys and string literals) that would be covered by this, and we should not have invalid JSON like that anywhere.

Reply:
The reason isLenient was used was because there was an unquoted number value the JSON string in data here https://github.com/powersync-ja/powersync-kotlin/blob/main/core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt#L357

Reply:
That would have caused Kotlin to treat it as a String instead of a number, which is a bug, and it's better to have a hard error for cases like that. That specific issue should be solved by the changes here (having `data` as a String instead of `Map<String, String>`).